docs/components/assembly.md

```diff
@@ -0,0 +1,50 @@
+# Assembly
+
+Assembles messages till criteria is met, the output will be the assembled message
+
+## Configuration Parameters
+
+```yaml
+component_name: <user-supplied-name>
+component_module: assembly
+component_config:
+  assemble_key: <string>
+  max_items: <string>
```

Comment:
I think this should be something like: number_of_items or required_items

'max' suggests that it would be acceptable to have fewer items that this number, which isn't true.

Reply:
I am thinking about this a bit more. My original idea for this component was that you are waiting for an exact number of pieces and if the timeout hit, it was an error. Maybe that isn't the case and it should be like aggregate, where it is okay if the timeout goes off without having received all the pieces. The next component could always produce an error if there weren't enough pieces in the flushed message...

Reply:
I personally think if we're handling some sort of buffer eg we're getting a buffer of live audio input and we want to flush the buffer when the user stops talking this would come in handy and as you mentioned the next component can always throw an error with a length check